src/openrct2/scripting/ScEntity.hpp

```diff
@@ -518,6 +519,23 @@ namespace OpenRCT2::Scripting
             }
         }
 
+        int16_t trackProgress_get() const
+        {
+            auto vehicle = GetVehicle();
+            return vehicle != nullptr ? vehicle->track_progress : 0;
+        }
+        void trackProgress_set(int16_t value)
+        {
+            ThrowIfGameStateNotMutable();
+            auto vehicle = GetVehicle();
+            if (vehicle != nullptr)
+            {
+                vehicle->Invalidate2();
```

Comment:
Do we need this first invalidate, as nothing has been performed yet?

Reply:
yes, otherwise it will only invalidate the screen at its new position and not its old position.